checks/shell_download_validate.go

```diff
@@ -768,7 +768,7 @@ func validateShellFileAndRecord(pathfn string, content []byte, files map[string]
 		// Note: this is caught by internal caller and only printed
 		// to avoid failing on shell scripts that our parser does not understand.
 		// Example: https://github.com/openssl/openssl/blob/master/util/shlib_wrap.sh.in
-		return false, sce.WithMessage(sce.ErrorShellParsing, err.Error())
+		return false, sce.WithMessage(sce.ErrorShellParsing, fmt.Sprintf("%s: %s", pathfn, err.Error()))
```

Comment:
Looking at https://pkg.go.dev/mvdan.cc/sh/v3/syntax#Parser.Parse, I think I should pass `pathfn` to `Parse` instead

Reply:
LGTM if you've verified it works.

Reply:
I certainly did :-) I'll rebase it once the integration test is fixed and https://github.com/ossf/scorecard/pull/1298 is merged.

FWIW the relevant "mvdan/sh" issue is https://github.com/mvdan/sh/issues/636. As far as I can tell it will never be fixed.